intro.js

```diff
@@ -864,10 +853,66 @@
       }
     }
 
+    // only top and bottom positions have optional alignments
+    if (['top', 'bottom'].indexOf(calculatedPosition) !== -1) {
+      calculatedPosition += _determineAutoAlignment(targetOffset.left, tooltipWidth, windowSize, desiredAlignment);
```

Comment:
alignment is calculated after position, when we know whether the auto-position will be top or bottom